Lib/multiprocessing/forkserver.py

```diff
@@ -165,8 +193,16 @@ def ensure_running(self):
 #
 #
 
-def main(listener_fd, alive_r, preload, main_path=None, sys_path=None):
-    '''Run forkserver.'''
+def main(listener_fd, alive_r, preload, main_path=None, sys_path=None,
+         *, authkey_r=None):
```

Comment:
What are the situations in which `authkey_r` would not be passed in?  Only in tests?

Reply:
realistically, I expect so.  I'm being overly cautious with the API change here as while `multiprocessing.forkserver` and `multiprocessing.forkserver.main` were never _intended_ to be APIs let alone public (thankfully not documented)... if anyone depends on them in gross poking at internals situations due to their naming this at least preserves existing semantics.  Even though our own use, the default, that 99.999% of anyone should be using, is to call this from `ensure_running()` above with `authkey_r=` always supplied as an argument.